Generator/JsonGenerator.cs

```diff
@@ -127,7 +127,8 @@ internal static void Generate(MetadataReader mr, string outDir)
             {
                 string filepath = Path.Combine(outDir, api.Name + ".json");
                 using var fileStream = new FileStream(filepath, FileMode.Create, FileAccess.Write, FileShare.Read);
-                using var streamWriter = new StreamWriter(fileStream, Encoding.UTF8);
+                // Encode the stream as UTF-8 without adding the byte order mark.
+                using var streamWriter = new StreamWriter(fileStream, new UTF8Encoding(false, true));

```

Comment:
Just a nitpick, should we use the constructor that only uses 1 bool?
```C#
                using var streamWriter = new StreamWriter(fileStream, new UTF8Encoding(false));
```

Reply:
My reasoning for using the two bool constructor was that any non-unicode input is likely a bug (probably upstream in the metadata) so I thought throwing an error makes some sense. However, I'm not confident that's the right approach and it is a separate concern from avoiding the BOM so I think I'll do as you suggest and use the 1 bool constructor here. This can always be revisited in a separate issue/PR (or not).